gdal/ogr/ogrfeature.cpp

```diff
@@ -2443,7 +2443,7 @@ const char *OGRFeature::GetFieldAsString( int iField ) const
     else if( eType == OFTTime )
     {
         const int ms = OGR_GET_MS(pauFields[iField].Date.Second);
-        if( ms != 0 )
+        if( ms != 0 || CPLIsNan(pauFields[iField].Date.Second) )
```

Comment:
couldn't you just swith the then and else clause and test ms == 0 ? This would avoid the annoying NaN check

Reply:
Ah that wouldn't solve the issue, as OGR_GET_MS() returns 0 for NaN. Ok merging thins then

Reply:
Thanks for thinking it through.  These trip me up so often.  This was my 3rd try at a patch for this issue.  My other tries were really ugly.